napari/_qt/layers/qt_image_base_layer.py

```diff
@@ -56,13 +59,83 @@ def __init__(self, layer):
     def changeColor(self, text):
         self.layer.colormap = text
 
-    def _on_clim_change(self, event=None):
+    def _clim_mousepress(self, event):
+        """when right clicking the clim slider, we want a QHRangeSliderPopup
+        to appear that provides greater control over the values and range of
+        the contrast limits.
+        """
+        if event.button() == Qt.RightButton:
+            precision = 2
+            try:
+                layer_is_int = np.issubdtype(self.layer.data.dtype, np.integer)
+                precision = 0 if layer_is_int else precision
+            except AttributeError:
+                layer_is_int = False
```

Comment:
another good catch.  The surface layer used to throw an Attribute error there (since `Surface.data` is a tuple.  But I later added the dtype property to surface making this obsolete.  thanks!